Sources/HAP/Base/Generated.swift

```diff
@@ -0,0 +1,6937 @@
+			super.init(type: .speaker, characteristics: unwrapped)
+		}
+	}
+
+	open class StatefulProgrammableSwitchBase: Service {
+		// Required Characteristics
+		public let programmableSwitchEvent: GenericCharacteristic<UInt8>
+		public let programmableSwitchOutputState: GenericCharacteristic<UInt8>
+
+		// Optional Characteristics
+		public let name: GenericCharacteristic<String>?
+
+		public init(characteristics: [AnyCharacteristic] = []) {
+			var unwrapped = characteristics.map { $0.wrapped }
+			programmableSwitchEvent = getOrCreateAppend(type: .programmableSwitchEvent, characteristics: &unwrapped, generator: { PredefinedCharacteristic.programmableSwitchEvent() })
+			programmableSwitchOutputState = getOrCreateAppend(type: .programmableSwitchOutputState, characteristics: &unwrapped, generator: { PredefinedCharacteristic.programmableSwitchOutputState() })
+			name = unwrapped.first { $0.type == .name } as? GenericCharacteristic<String>
+			super.init(type: .statefulProgrammableSwitch, characteristics: unwrapped)
+		}
+	}
+
+	open class StatelessProgrammableSwitchBase: Service {
+		// Required Characteristics
+		public let programmableSwitchEvent: GenericCharacteristic<UInt8>
+
+		// Optional Characteristics
+		public let name: GenericCharacteristic<String>?
+		public let labelIndex: GenericCharacteristic<UInt8>?
+
+		public init(characteristics: [AnyCharacteristic] = []) {
+			var unwrapped = characteristics.map { $0.wrapped }
+			programmableSwitchEvent = getOrCreateAppend(type: .programmableSwitchEvent, characteristics: &unwrapped, generator: { PredefinedCharacteristic.programmableSwitchEvent() })
+			name = unwrapped.first { $0.type == .name } as? GenericCharacteristic<String>
+			labelIndex = unwrapped.first { $0.type == .labelIndex } as? GenericCharacteristic<UInt8>
+			super.init(type: .statelessProgrammableSwitch, characteristics: unwrapped)
+		}
+	}
+
+	open class SwitchBase: Service {
+		// Required Characteristics
+		public let powerState: GenericCharacteristic<Bool>
+
+		// Optional Characteristics
+		public let name: GenericCharacteristic<String>?
+
+		public init(characteristics: [AnyCharacteristic] = []) {
+			var unwrapped = characteristics.map { $0.wrapped }
+			powerState = getOrCreateAppend(type: .powerState, characteristics: &unwrapped, generator: { PredefinedCharacteristic.powerState() })
+			name = unwrapped.first { $0.type == .name } as? GenericCharacteristic<String>
+			super.init(type: .`switch`, characteristics: unwrapped)
+		}
+	}
+
+	open class TelevisionBase: Service {
+		// Required Characteristics
+		public let active: GenericCharacteristic<Enums.Active>
+		public let activeIdentifier: GenericCharacteristic<UInt32>
+		public let configuredName: GenericCharacteristic<String>
+		public let sleepDiscoveryMode: GenericCharacteristic<Enums.SleepDiscoveryMode>
+
+		// Optional Characteristics
+		public let brightness: GenericCharacteristic<Int>?
+		public let closedCaptions: GenericCharacteristic<Enums.Close
```

Comment:
Todo: set default values

Reply:
I've found that `Lightbulb` from the hap-server example doesn't work anymore, due to the powerstate not having a value. I think we should set the value for all characteristics with read permission to something; e.g. the default value for that type:

* int -> minValue or 0
* bool -> false
* etc..